internal/controllers/utils/utils.go

```diff
@@ -1218,3 +1256,77 @@ func MapKeysToSlice(inputMap map[string]bool) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+// OAuthClientConfig defines the parameters required to establish an HTTP Client capable of acquiring an OAuth Token
+// from an OAuth capable authorization server.
+type OAuthClientConfig struct {
```

Comment:
I think this struct could be moved at the top of the file, together with the other vars/const?

Reply:
fixed